src/openapi-generator.ts

```diff
@@ -85,15 +88,22 @@ export class OpenAPIGenerator {
     component: OpenAPIComponentObject;
   }[] = [];
 
-  constructor(private definitions: OpenAPIDefinitions[]) {
+  constructor(
+    private definitions: OpenAPIDefinitions[],
+    private openAPIVersion: OpenApiVersion = '3.0.0'
+  ) {
     this.sortDefinitions();
   }
 
   generateDocument(config: OpenAPIObjectConfig): OpenAPIObject {
+    if (config.openapi) {
+      this.openAPIVersion = config.openapi;
```

Comment:
Hmm this seems both correct and incorrect.
1. I feel like we should not be modifying the value provided in the constructor. It's quite implicit. And I would be okay for the user to "break his head" at his own risk
2. I do get your point that we should try and "keep him from breaking his head"

My gut feeling is more inclined to go with 1. The best example of why I prefer it is something like the following:
```ts
const generator = new OpenAPIGenerator(schemas, '3.0.0');
generator.generateComponents(); // this would have output for 3.0.0
generator.generateDocument({ openapi: '3.1.0', ... });
generator.generateComponents(); // this would have output for 3.1.0
```
The reason for the change in behavior of `generateComponents` is not really obvious.
cc: @georgyangelov  any opinion?

Reply:
Yeah okay that makes sense to me. Wasn't sure if we wanted to go with breaking change or not but not mutating the constructor's version makes sense.